src/measure_theory/covering/density_theorem.lean

```diff
@@ -0,0 +1,148 @@
+example we want hyperbolic space to carry the instance `is_doubling_measure volume` but volumes grow
+exponentially in hyperbolic space. To be really explicit, consider the hyperbolic plane of
+curvature -1, the area of a disc of radius `ε` is `A(ε) = 2π(cosh(ε) - 1)` so `A(2ε)/A(ε) ~ exp(ε)`.
+-/
+class is_doubling_measure {α : Type*} [metric_space α] [measurable_space α] (μ : measure α) :=
+(exists_measure_closed_ball_le_mul [] :
+  ∃ (C : ℝ≥0), ∀ᶠ ε in 𝓝[>] 0, ∀ x, μ (closed_ball x (2 * ε)) ≤ C * μ (closed_ball x ε))
+
+namespace is_doubling_measure
+
+variables {α : Type*} [metric_space α] [measurable_space α] (μ : measure α) [is_doubling_measure μ]
+
+/-- A doubling constant for a doubling measure. -/
+def doubling_constant : ℝ≥0 := classical.some $ exists_measure_closed_ball_le_mul μ
+
+lemma exists_measure_closed_ball_le_mul' :
+  ∀ᶠ ε in 𝓝[>] 0, ∀ x, μ (closed_ball x (2 * ε)) ≤ (doubling_constant μ) * μ (closed_ball x ε) :=
+classical.some_spec $ exists_measure_closed_ball_le_mul μ
+
+lemma exists_eventually_forall_measure_closed_ball_le_mul (K : ℝ) :
+  ∃ (C : ℝ≥0), ∀ᶠ ε in 𝓝[>] 0, ∀ x t (ht : t ≤ K),
+    μ (closed_ball x (t * ε)) ≤ C * μ (closed_ball x ε) :=
+begin
+  let C := doubling_constant μ,
+  have hμ : ∀ (n : ℕ), ∀ᶠ ε in 𝓝[>] 0, ∀ x,
+    μ (closed_ball x (2^n * ε)) ≤ ↑(C^n) * μ (closed_ball x ε),
+  { intros n,
+    induction n with n ih, { simp, },
+    replace ih := eventually_nhds_within_pos_mul_left (two_pos : 0 < (2 : ℝ)) ih,
+    refine (ih.and (exists_measure_closed_ball_le_mul' μ)).mono (λ ε hε x, _),
+    calc μ (closed_ball x (2^(n + 1) * ε))
+          = μ (closed_ball x (2^n * (2 * ε))) : by rw [pow_succ', mul_assoc]
+      ... ≤ ↑(C^n) * μ (closed_ball x (2 * ε)) : hε.1 x
+      ... ≤ ↑(C^n) * (C * μ (closed_ball x ε)) : ennreal.mul_left_mono (hε.2 x)
+      ... = ↑(C^(n + 1)) * μ (closed_ball x ε) : by rw [← mul_assoc, pow_succ', ennreal.coe_mul], },
+  rcases lt_or_le K 1 with hK | hK,
+  { refine ⟨1, _⟩,
+    simp only [ennreal.coe_one, one_mul],
+    exact eventually_mem_nhds_within.mono (λ ε hε x t ht,
+      measure_mono $ closed_ball_subset_closed_ball (by nlinarith [mem_Ioi.mp hε])), },
+  { refine ⟨C^⌈real.logb 2 K⌉₊, ((hμ ⌈real.logb 2 K⌉₊).and eventually_mem_nhds_within).mono
+      (λ ε hε x t ht, le_trans (measure_mono $ closed_ball_subset_closed_ball _) (hε.1 x))⟩,
+    refine mul_le_mul_of_nonneg_right (ht.trans _) (mem_Ioi.mp hε.2).le,
+    conv_lhs { rw ← real.rpow_logb two_pos (by norm_num) (by linarith : 0 < K), },
+    rw ← real.rpow_nat_cast,
+    exact real.rpow_le_rpow_of_exponent_le one_le_two (nat.le_ceil (real.logb 2 K)), },
+end
+
+variables [proper_space α] [borel_space α] [is_locally_finite_measure μ]
+
+/-- A Vitali family in space with doubling measure with a covering proportion controlled by `K`. -/
+def vitali_family (K : ℝ) (hK : 6 ≤ K) : vitali_family μ :=
+vitali.vitali_family μ
+  (classical.some (exists_eventually_forall_measure_closed_ball_le_mul μ K)) $ λ x ε hε,
```

Comment:
Maybe you want `doubling_constant' K`?

Reply:
Good point; I've gone with `scaling_constant_of`.